design-docs/improved-console.md

```diff
@@ -0,0 +1,446 @@
+* Renders well on default macOS Terminal, default Windows Console, Cygwin, PowerShell and
+ common Linux Terminals
+* Console shows idle workers under `--continuous` build
+
+## Story: Display overall build progress as a progress bar
+Render ProgressEvents with build completeness information as a progress bar.
+ 
+Intended to give the user a very fast way of telling whether the build will be finished soon or not.
+
+### User-visible Changes
+We can visually represent complete and un-started tasks of build using colorized output:
+
+`###############>                       > 40% Building`
+
+On ANSI terminals, we can use empty spaces with different background colors for the "bars".
+The progress bar shall be red given a failing build.
+
+Note that the suffix at the end of the progress bar should be the progress for the current
+Gradle execution phase.
+
+**ASCII-based options (some options use extended ASCII):**
+```
+[##                        ] 6% Building
+[‡‡‡‡                      ] 12% Building
+###########>               > 33% Building
+‹===================       › 60% Building
+«==========================» 100% BUILD SUCCESS
+```
+
+### Implementation
+`DefaultGradleLauncherFactory` registers a listener `BuildProgressBarListener` (similar to `BuildProgressFilter`)
+and forwards events to a `ConsoleBackedProgressBarRenderer`. It would maintain a `BuildProgressBarLogger` that
+logs using a `ProgressBar` implementation of `ProgressFormatter`. A `ProgressBar` has a width and filler character.
+
+`ConsoleBackedProgressBarRenderer` renders the `ProgressBar` using a `ProgressLabel`, which extends `Label` with 
+the following additions:
+ - `void setProgressBar(ProgressBar progressBar)`
+ - `void setText(String text)` (inherited from `Label`)
+ - `void render()` (we can probably add this to `Label`)
+ 
+Under `--continuous` build, the progress bar and multi status text area are reused.
+
+User guide is updated with new output examples.
+
+### Test Coverage
+* By default, logs should be streamed with plain output and not throttled when not attached to a Console
+* Color output is output when requested through via the tooling API via `LongRunningOperation.setColorOutput(true)`
+* OutputEvents tooling to API is unaffected except for additional BuildOperations now published
+* Logs should be streamed with plain output when attached Terminal lacks of color or cursor support
+* Renders well on default macOS Terminal, default Windows Console, Cygwin, PowerShell and common Linux Terminals
+
+## Story: Improve BuildOperation name consistency
+Standardize the terminology and format of build operation names. This is necessary for a cohesive
+experience as these are displayed while the operation is executing. 
+
+### TODO Implementation
+
+### Test Coverage
+* Update tests to reflect updated output where necessary
+
+### Open Issues
+* How is buildSrc rendered?
+* How are composite builds rendered?
+* How are GradleBuild tasks rendered?
```

Comment:
For these, the answer is: same as everything else. Build operation events are generated for these builds.